transcom/mymove
# B-20947: SSW Actual Obligations 100% GCC Calculations

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20947)

Merge after B-20946, as this has work built off of code on that branch
## [B-20946 INT PR](https://github.com/transcom/mymove/pull/13573)
## [B-20948 INT PR](https://github.com/transcom/mymove/pull/13590)


## Summary

The SSW Actual Obligations 100% GCC Calculation was wrong because it was using the estimated weight. I changed the calculation to use the Actual Weight (that comes from the weight tickets) up to the maximum weight entitlement or allowable weight, whichever is lessor.

Actual Weight for a PPM is calculated by function getActualWeight that does all of the work. This unction adds up the total weight for the weight tickets and returns its value. functionPPMCloseoutFetcher.GetActualWeight(ppmShipment) at pkg/services/ppm_closeout/ppm_closeout.go line 227

You can also see the actual weight on the MTO tab as a counselor. "Actual PPM Weight"
![image](https://github.com/user-attachments/assets/4daefda0-7f0b-4ee6-8f64-228e2935aa68)


Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access office as a service counselor
2. Select a PPM Closeout move
3. while under Move Details Tab scroll down and select "Download AOA Paperwork (PDF)"
4. Once downloaded inspect the 100% GCC value under Actual Obligations section under the Entitlements & move summary
5. The value should be the actual weight up to the (entitlement weight or allowed weight) whichever is lessor. The value should also not include any Progear. For Example:

If:
Entitlement = 5000
TotalWeight = 7500
Actual Weight = 4000
Then 100% GCC should equal 4000lbs

If:
Entitlement = 5000
TotalWeight = 7500
Actual Weight = 6000
Then 100% GCC should equal 5000lbs


You can add weight tickets as a customer by going to a customer with a PPM Shipment and following step 6:
![image](https://github.com/user-attachments/assets/a5e6034b-6141-4595-a96c-4bba7c43cdf6)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![image](https://github.com/user-attachments/assets/d32144ea-ffd5-47b3-b9a1-30c3faf5fd37)
